frontend/src/pages/instance/Performance.vue

```diff
@@ -0,0 +1,387 @@
+            }
+
+            // If httpAdminRoot is not set (or '/'), then we can use launcher 1.13 or later.
+            // Otherwise, we need launcher 2.18 or later due to a bug fix for handling httpAdminRoot in the resource API
+            let minVersion = '>=1.13'
+            if (this.instance?.settings?.httpAdminRoot && this.instance.settings.httpAdminRoot !== '/') {
+                minVersion = '>=2.18.0'
+            }
+            const nrLauncherVersion = SemVer.coerce(this.instance?.meta?.versions?.launcher)
+            return SemVer.satisfies(nrLauncherVersion, minVersion)
+        },
+        featureAvailable () {
+            return this.isInstanceResourcesFeatureEnabledForPlatform &&
+                this.isInstanceResourcesFeatureEnabledForTeam &&
+                this.launcherSatisfiesVersion
+        }
+    },
+    mounted () {
+        if (this.featureAvailable) {
+            this.getResources()
+                .then(() => {
+                    this.connectToLiveData()
+                })
+                .catch(e => {
+                    this.error = e
+                })
+                .finally(() => {
+                    this.loading = false
+                })
+        } else {
+            this.loading = false
+        }
+    },
+    methods: {
+        getResources () {
+            if (this.isInstanceRunning) {
+                return instancesApi.getResources(this.instance.id)
+                    .then(response => {
+                        if (response.resources.length === 0) {
+                            this.error = 'Waiting for resource data'
+                        } else if (response.resources.length > 0) {
+                            let hasCPU = false
+                            // sanity check we have "cpu" in the resources
+                            for (const resource of response.resources) {
+                                if (Object.prototype.hasOwnProperty.call(resource, 'cpu')) {
+                                    hasCPU = true
+                                    break
+                                }
+                            }
+                            if (!hasCPU) {
+                                this.error = 'CPU Utilization data is not available for this instance'
+                            } else {
+                                this.resources = response.resources
+                            }
+                        } else {
+                            this.error = null
+                        }
+                    })
+                    .catch(e => {
+                        this.error = e
+                    })
+            }
+
+            return Promise.reject(new Error('Instance is not running.'))
+        },
+        connectToLiveData () {
+            if (this.wsConnected) {
+                return
+            }
+            const uri = `/api/v1/projects/${this.instance.id}/resources/stream`
+            const ws = new WebSocket(uri)
```

Comment:
This websocket is never closed when the user navigates away from the page.

Reply:
Handled in https://github.com/FlowFuse/flowfuse/pull/5528/commits/bd474295f5c73f0a8601997f560034eb116a19c6